sonic_platform_base/sonic_thermal_control/thermal_action_base.py

```diff
@@ -0,0 +1,72 @@
+class ThermalActionBase(object):
+    """
+    Base class for thermal action. Once all thermal conditions in a thermal policy are matched,
+    all predefined thermal action would be executed.
+    """
+    # JSON field definition
+    JSON_FIELD_ACTION_TYPE = 'type'
+
+    # Dictionary of ThermalActionBase-derived class representing all thermal action types
+    _action_type_dict = {}
+
+    def execute(self, thermal_info_dict):
+        """
+        Take action when thermal condition matches. For example, adjust speed of fan or shut
+        down the switch.
+        :param thermal_info_dict: A dictionary stores all thermal information.
+        :return:
+        """
+        raise NotImplementedError
+
+    def load_from_json(self, json_obj):
```

Comment:
> this function name is too ambiguous, rename it like "PolicyParser" something like this.

Actually, load_from_json is kind of common name to describe a function that initialize or de-serialize an object from JSON. Since we are de-serialize an thermal action object from policy, maybe we can call this function "parse_policy", how do you think?